cxxheaderparser/parser.py

```diff
@@ -2076,6 +2078,29 @@ def _parse_function(
                 self.visitor.on_method_impl(state, method)
 
             return method.has_body or method.has_trailing_return
+        elif is_guide:
+            fn = Function(
+                Type(PQName([AutoSpecifier()])),
+                pqname,
+                params,
+                vararg,
+                doxygen=doxygen,
+                template=template,
+                operator=op,
+                **props
+            )
+            if not self.lex.token_if("ARROW"):
+                raise self._parse_error(None, expected="Trailing return type")
+            self._parse_trailing_return_type(fn)
```

Comment:
Since this function can't fail, I think it would be better to refactor this to be `_parse_trailing_return_type(self, Optional[DecoratedType]) -> DecoratedType`, and let the caller assign the returned type to the correct structure. Then you don't need to abuse Function here.

Reply:
That works, I've pushed an update that does exactly that. Thanks!